amundsen_application/models/user.py

```diff
@@ -63,19 +63,19 @@ class UserSchema(Schema):
     role_name = fields.Str(allow_none=True)
     slack_id = fields.Str(allow_none=True)
     team_name = fields.Str(allow_none=True)
-    user_id = fields.Str(required=True)
+    user_id = fields.Str(allow_none=False)
 
     @pre_load
     def preprocess_data(self, data: Dict) -> Dict:
-        if not data.get('user_id', None):
-            data['user_id'] = data.get('email', None)
+        if _str_no_value(data.get('user_id')):
+            data['user_id'] = data.get('email')
 
-        if not data.get('profile_url', None):
+        if _str_no_value(data.get('profile_url')):
             data['profile_url'] = ''
             if app.config['GET_PROFILE_URL']:
                 data['profile_url'] = app.config['GET_PROFILE_URL'](data['user_id'])
 
-        if not data.get('display_name', None):
+        if _str_no_value(data.get('display_name')):
             data['display_name'] = data.get('full_name', data.get('email'))
```

Comment:
`data.get('full_name', data.get('email'))` is the root cause of the bug, when `full_name` is an empty string, it assigns that value to `data['display_name']`. Perhaps something like:

```
if not data.get('display_name'):
    if data.get('full_name'):
        data['display_name'] = data.get('full_name')
    else:
        data['display_name'] = data.get('email')
```
_disclaimer: this code could probably be cleaned up_